memory/org.eclipse.cdt.debug.ui.memory.search/src/org/eclipse/cdt/debug/ui/memory/search/MemorySearchResultsPage.java

```diff
@@ -173,34 +174,48 @@ public Object[] getElements(Object inputElement) {
-					IMemoryRenderingContainer containers[] = ((IMemorySearchQuery) fQuery).getMemoryView()
-							.getMemoryRenderingContainers();
+				IMemoryRenderingContainer containers[] = ((IMemorySearchQuery) fQuery).getMemoryView()
+						.getMemoryRenderingContainers();
+				if (containers == null || containers.length == 0)
+					return;
+
+				if (event.getSelection() instanceof StructuredSelection
+						&& ((StructuredSelection) event.getSelection()).getFirstElement() instanceof MemoryMatch) {
+
 					MemoryMatch match = (MemoryMatch) ((StructuredSelection) event.getSelection()).getFirstElement();
-					if (match != null) {
-						for (int i = 0; i < containers.length; i++) {
-							IMemoryRendering rendering = containers[i].getActiveRendering();
-							if (rendering instanceof IRepositionableMemoryRendering) {
-								try {
-									((IRepositionableMemoryRendering) rendering).goToAddress(match.getStartAddress());
-								} catch (DebugException e) {
-									MemorySearchPlugin.logError(
-											Messages.getString("MemorySearchResultsPage.RepositioningMemoryViewFailed"), //$NON-NLS-1$
-											e);
-								}
-							}
-
-							if (rendering != null) {
-								// Temporary, until platform accepts/adds new interface for setting the selection
-								try {
-									Method m = rendering.getClass().getMethod("setSelection", //$NON-NLS-1$
-											new Class[] { BigInteger.class, BigInteger.class });
-									if (m != null)
-										m.invoke(rendering, match.getStartAddress(), match.getEndAddress());
-								} catch (Exception e) {
-									// do nothing
-								}
-							}
+					if (match == null)
+						return;
+
+					for (int i = 0; i < containers.length; i++) {
+						IMemoryRendering rendering = containers[i].getActiveRendering();
+						if (rendering == null || !(rendering instanceof IRepositionableMemoryRendering))
+							continue;
+
+						try {
+							((IRepositionableMemoryRendering) rendering).goToAddress(match.getStartAddress());
+						} catch (DebugException e) {
+							MemorySearchPlugin.logError(
+									Messages.getString("MemorySearchResultsPage.RepositioningMemoryViewFailed"), //$NON-NLS-1$
+									e);
+						}
+
+						// Open the memory view to emphasize the effect of the address selection in the search view
+						IWorkbenchPart wb = containers[i].getMemoryRenderingSite().getSite().getPart();
+						if (wb == null)
+							continue;
+
+						getSite().getPage().activate(wb);
+
+						// Temporary, until platform accepts/adds new interface for setting the selection
+						try {
+							Method m = rendering.getClass().getMethod("setSelection", //$NON-NLS-1$
+									new Class[] { BigInteger.class, BigInteger.class });
+							if (m != null)
+								m.invoke(rendering, match.getStartAddress(), match.getEndAddress());
+						} catch (Exception e) {
+							MemorySearchPlugin.logError(
+									"Exception while invoking the setSelection method for the current rendering", //$NON-NLS-1$
+									e);
```

Comment:
This is now filling my log with errors everytime I move around the selection. I do agree that logging Exception is a good idea, but there should remain an empty `catch (NoSuchMethodException e1)` so that that expected error isn't logged.

PS I'm really not keen on the reflection use at all here and would love to see that fixed as it clearly isn't "temporary"

```suggestion
						} catch (NoSuchMethodException e) {
							// Not all renderings have a compatible setSelection
							// ideally this would be an interface we can check instead
							// of using reflection!
						} catch (Exception e) {
							MemorySearchPlugin.logError(
									"Exception while invoking the setSelection method for the current rendering", //$NON-NLS-1$
									e);
```